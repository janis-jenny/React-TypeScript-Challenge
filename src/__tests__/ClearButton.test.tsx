import renderer from 'react-test-renderer';
import Stack from '@mui/material/Stack';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from '@material-ui/core/Button';

describe('ClearButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <>
        <Stack><Button /></Stack>
      </>,
    );
    expect(tree).toMatchSnapshot();
  });
});

test('1. ClearButton Rendering', () => {
  let buttons = document.getElementsByTagName('button');
  expect(buttons.length).toBe(0);
  render(<Button type="button" />);
  buttons = document.getElementsByTagName('button');
  expect(buttons.length).toBe(1);
});

test('2. Handles onClick', () => {
  const onClick = jest.fn();
  render(<Button type="button" onClick={onClick}/>);
  const btnElement = screen.getByRole('button');
  fireEvent.click(btnElement);
  expect(onClick).toHaveBeenCalledTimes(1);
});
