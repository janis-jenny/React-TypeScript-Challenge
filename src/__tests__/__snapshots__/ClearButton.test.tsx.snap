// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton renders correctly 1`] = `
<div
  className="css-nen11g-MuiStack-root"
>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  />
</div>
`;
